src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,46 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  books: [
+    {
+      item_id: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
+};
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, actions) => {
+      const bookData = actions.payload;
+      state.books.push(bookData);
```

Comment:
- [ ] You have the book you need to add so you need to return a new state by making a copy of the state of the books as well as the book you’re updating

  for instance :point_down: 
  ```javascript
   addBook: (state, action) => {
        state.books = [
          ...state.books, action.payload
        ]
      }
  ```